tiledb/sm/filesystem/vfs.cc

```diff
@@ -1707,4 +1712,60 @@ Status VFS::flush_multipart_file_buffer(const URI& uri) {
   return Status::Ok();
 }
 
+void VFS::log_read(const URI& uri, uint64_t offset, uint64_t nbytes) {
+  std::string fragments_component = "/__fragments/";
```

Comment:
It would be beneficial not to continue the practice of scattering constant strings regarding our storage format throughout the code. It looks like this one could be replaced by a function `tiledb::sm::utils::parse::is_fragment()` and another one that returns whatever's needed by the `FRAGMENTS` case.

Reply:
Fair enough.

Reply:
I ended up going with `std::optional<std::string> URI::get_fragment_name();` instead of `tiledb::sm::utils::parse::is_fragment()` for two reasons. First, `get_fragment_name` is really what I need since I want to log/dedupe at the fragment level on the least verbosity end of the spectrum. Second, it went in the URI class because when I added it to `parse_uri.cc` I got to around 30 different CMakeLists.txt updates across the entire build system before I abandoned that approach.

Reply:
What's in the new code is good enough for now. It's in `class URI`, which is the wrong place, but right place is something like `class ArrayURI` which doesn't exist. (We don't have fragments in group URI, for example.) Having it in `class URI`, though, is only a little bit wrong, nowhere near digging-the-hole-deeper wrong.